torch/_dynamo/side_effects.py

```diff
@@ -209,7 +210,7 @@ def has_pending_mutation_of_attr(self, item, name):
     def is_modified(self, item):
         if item.is_immutable():
             return False
-        if isinstance(item.mutation_type, AttributeMutationNew):
+        if isinstance(item.mutation_type, (AttributeMutationNew, ValueMutationNew)):
```

Comment:
This one is a bit surprising, since `ValueMutationNew` suggests we don't need to track it for side effects, e.g., it's a dictionary/list created by the code Dynamo is tracing through, and its state would be fully represented by `ListVariable` or `ConstDictVariable`.

It's possible that we attached the wrong `mutation_type` somewhere, or I might be missing something.